File: src/js/src/satisfiers/CannotNestInsideInlineSatisfier.js
function CannotNestInsideInlineSatisfier() { 
	this.isSatisfiedBy = function(context) {
		return !any(context.scopes, function(scope) {
			return scope.level === ScopeLevel.inline;
		});
	};
}

extend(Satisfier, CannotNestInsideInlineSatisfier);

File: src/dotnet/Src/ButterflyNet.Parser/Satisfiers/CannotNestInsideInlineSatisfier.cs
using System.Linq;

namespace ButterflyNet.Parser.Satisfiers {
	public class CannotNestInsideInlineSatisfier : ISatisfier {
		public bool IsSatisfiedBy(ParseContext context) {
			return !context.Scopes.Any(scope => scope.Level == ScopeLevel.Inline);
		}
	}
}